File: plugins/flutter_bg_location_plugin/android/src/main/kotlin/com/ezride/flutter_bg_location_plugin/handlers/LocationServiceStartHandler.kt
package com.ezride.flutter_bg_location_plugin.handlers

import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import android.content.Context
import android.content.Intent
import android.util.Log
import com.ezride.flutter_bg_location_plugin.services.LocationService

class LocationServiceStartHandler : Handler{


    override val callMethod : String = CallMethods.LOCATION_SERVICE_START;

    override fun handler(context: Context, call: MethodCall, result: MethodChannel.Result){
        Log.d("FlutterLocationPlugin", "startTracking invoked")
        // Получаем параметр seconds и hash из аргументов. это время для сессии трекинга.
        val args = call.arguments as? Map<*, *>
        val seconds = (args?.get("seconds") as? Int) ?: 0;
        val hash = (args?.get("hash") as? String) ?: "";
        val orderId = (args?.get("orderId") as? Int) ?: 0;
        val isStarted =  LocationService.startTracking(context,seconds,hash,orderId);
        result.success(isStarted);

    }
}

File: plugins/flutter_bg_location_plugin/ios/Classes/Handler/LocationServiceStartHandler.swift
import  Flutter

final class LocationServiceStartHandler: Handler {

    let callMethod = CallMethods.LOCATION_SERVICE_START

    func handler(call: FlutterMethodCall, result: @escaping FlutterResult) {
        print("FlutterLocationPlugin", "startTracking invoked")
        guard let args = call.arguments as? [String: Any] else { return result(false) }

        let seconds = args["seconds"] as? Int ?? 0;
        let hash = args["hash"] as? String ?? "";
        let orderId = args["orderId"] as? Int ?? 0;
  
        let isStarted = PluginContext.shared.locationService.startTracking(seconds:seconds, hash:hash, orderId:orderId)       
        result(isStarted)
    }
}
